python/cudf/cudf/core/index.py

```diff
@@ -203,6 +203,37 @@ def _mimic_inplace(self, other, inplace=False):
         else:
             return other
 
+    def __iter__(self):
+        raise TypeError(
+            "Creation of an Iterator over a cudf.Index is not allowed, "
+            "To create an iterator, explicitly convert to any of the objects "
+            "supporting iteration using .to_pandas(), "
+            ".to_arrow(), .values_host."
+        )
+
+    @property
+    def values_host(self):
+        """
+        Return a numpy representation of the Index.
+
+        Only the values in the Index will be returned.
+
+        Returns
+        -------
+        out : numpy.ndarray
+            The values of the Series.
+
+        Examples
+        --------
+        >>> import cudf
+        >>> index = cudf.Index([1, -10, 100, 20])
+        >>> index.values_host
+        array([  1, -10, 100,  20])
+        >>> type(index.values)
+        <class 'numpy.ndarray'>
+        """
+        return self.to_series().values_host
```

Comment:
Should move `.values_host` to the Column class and use it via that from both Index and Series.

Reply:
Moved both `.values` & `.values_host` to column level